internal/services/compute/image_resource.go

```diff
@@ -195,6 +196,22 @@ func resourceImage() *pluginsdk.Resource {
 			"tags": commonschema.Tags(),
 		},
 	}
+
+	if features.FourPointOhBeta() {
+		storageTypeProperty := &pluginsdk.Schema{
+			Type:         pluginsdk.TypeString,
+			Description:  "The type of storage disk",
+			Optional:     true,
+			ForceNew:     true,
+			Default:      string(images.StorageAccountTypesStandardLRS),
+			ValidateFunc: validation.StringInSlice(images.PossibleValuesForStorageAccountTypes(), false),
+		}
+
+		resource.Schema["os_disk"].Elem.(*pluginsdk.Resource).Schema["storage_type"] = storageTypeProperty
+		resource.Schema["data_disk"].Elem.(*pluginsdk.Resource).Schema["storage_type"] = storageTypeProperty
+	}
```

Comment:
Could we invert the condition here, i.e. adding the new property to the schema above, and having something like the following to conditionally remove it in 3.x mode? This will make it easier to update when 4.0 is released.

```suggestion
	if !features.FourPointOhBeta() {
		delete(resource.Schema["os_disk"].Elem.(*pluginsdk.Resource).Schema, "storage_type")
		delete(resource.Schema["data_disk"].Elem.(*pluginsdk.Resource).Schema, "storage_type")
	}
```

Reply:
That's a much better implementation, it's so obvious looking at it here 😄